src/ProblemSet.hs

```diff
@@ -21,35 +23,35 @@ reference :: String -> String -> StdGen -> String
 reference topic sit g = topic ++ "." ++ sit ++ " " ++ show g
 
 
-generate :: Int -> [Topic] -> StdGen -> IO ([SituationInstance], StdGen)
-generate 0 _      g = return ([], g)
-generate n topics g = let
-    makeInst = do
-        topic <- pickItem topics
-        gen <- get
-        situation <- choose topic
-        let ref = reference (refName topic) (sitRef situation) gen
-        instantiate ref situation
-    (sitInst, g') = runState makeInst g
-  in do
-    when (n `mod` 10 == 0) (putStr "." >> hFlush stdout)
-    maybeSit <- sitInst
+addStack :: Monad m => State StdGen a -> StateT StdGen m a
+addStack = mapStateT (return . runIdentity)
```

Comment:
I wonder if there's an easy way to get rid of this part; it feels like it ought to be unnecessary if I had my types right.